File: gaia-sdk-python/gaia_sdk/graphql/request/input/DeleteRoleImpulse.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class DeleteRoleImpulse():
    """
    The specification to delete a role instance
    """
    tenantId: str
    roleId: str

    def __init__(self, tenantId: str, roleId: str):
        self.tenantId = tenantId
        self.roleId = roleId

    def __eq__(self, other):
        if type(other) is type(self):
            return self.tenantId == other.tenantId and self.roleId == other.roleId
        return False

    def __repr__(self):
        return {'tenantId': self.tenantId, 'roleId': self.roleId}

File: gaia-sdk-javascript/src/graphql/request/input/DeleteRoleImpulse.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * The specification to delete a role instance
 */
export class DeleteRoleImpulse {
public _typeName = "DeleteRoleImpulse";
    private tenantId:Uuid;
    private roleId:Uuid;

    constructor (tenantId:Uuid, roleId:Uuid) {
        this.tenantId = tenantId;
        this.roleId = roleId;
    }
}
